### PR TITLE
[CycleGAN Tutorial] Refine the cycle loss function, and remove redundant gradient tape

### PR DESCRIPTION
**Pull request by junghau**
- Separating the cycle loss for F(G(x)) and G(F(y)) results in the
gradient not calculated on the loss of G(F(y)) with respect to G, and
loss of F(G(x)) with respect to F. Therefore they are now combined.

- Redundant gradient tape is removed since persistent is already set to
true.

### CODE REVIEW

**Comment by junghau** [comment]
I have just signed the Contributor License Agreement.

File: site/en/r2/tutorials/generative/cyclegan.ipynb [hunk 1]
```diff
@@ -84,8 +112,8 @@
         "\n",
         "This opens up the possibility to do a lot of interesting tasks like photo-enhancement, image colorization, style transfer, etc. All you need is the source and the target dataset (which is simply a directory of images).\n",
         "\n",
-        "![Output Image 1](images/horse2zebra_1.png)\n",
-        "![Output Image 2](images/horse2zebra_2.png)"
+        "![Output Image 1](https://github.com/junghau/docs/blob/better-cyclegan-loss/site/en/r2/tutorials/generative/images/horse2zebra_1.png?raw=1)\n",
```
Comment (lamberta) [inline site/en/r2/tutorials/generative/cyclegan.ipynb]: Undo change to image path.

If this doesn't work for some reason, then use the tensorflow/docs repo.

File: site/en/r2/tutorials/generative/cyclegan.ipynb [hunk 2]
```diff
@@ -402,23 +430,21 @@
...
         "There are 2 generators (G and F) and 2 discriminators (X and Y) being trained here. \n",
-        "* Generator `G` learns to transform image `X` to image `Y`. $(G: X -\u003e Y)$\n",
-        "* Generator `F` learns to transform image `Y` to image `X`. $(F: Y -\u003e X)$\n",
+        "* Generator `G` learns to transform image `X` to image `Y`. $(G: X -> Y)$\n",
+        "* Generator `F` learns to transform image `Y` to image `X`. $(F: Y -> X)$\n",
         "* Discriminator `D_X` learns to differentiate between image `X` and generated image `X` (`F(Y)`).\n",
         "* Discriminator `D_Y` learns to differentiate between image `Y` and generated image `Y` (`G(X)`).\n",
         "\n",
-        "![Cyclegan model](images/cyclegan_model.png)"
+        "![Cyclegan model](https://github.com/junghau/docs/blob/better-cyclegan-loss/site/en/r2/tutorials/generative/images/cyclegan_model.png?raw=1)"
```
Comment (lamberta) [inline site/en/r2/tutorials/generative/cyclegan.ipynb]: Undo change to image path

File: site/en/r2/tutorials/generative/cyclegan.ipynb [hunk 3]
```diff
@@ -576,29 +604,29 @@
...
         "\n",
-        "$$forward\\ cycle\\ consistency\\ loss: X -\u003e G(X) -\u003e F(G(X)) \\sim \\hat{X}$$\n",
+        "$$forward\\ cycle\\ consistency\\ loss: X -> G(X) -> F(G(X)) \\sim \\hat{X}$$\n",
         "\n",
-        "$$backward\\ cycle\\ consistency\\ loss: Y -\u003e F(Y) -\u003e G(F(Y)) \\sim \\hat{Y}$$\n",
+        "$$backward\\ cycle\\ consistency\\ loss: Y -> F(Y) -> G(F(Y)) \\sim \\hat{Y}$$\n",
         "\n",
         "\n",
-        "![Cycle loss](images/cycle_loss.png)"
+        "![Cycle loss](https://github.com/junghau/docs/blob/better-cyclegan-loss/site/en/r2/tutorials/generative/images/cycle_loss.png?raw=1)"
```
Comment (lamberta) [inline site/en/r2/tutorials/generative/cyclegan.ipynb]: Undo change to image path

File: site/en/r2/tutorials/generative/cyclegan.ipynb [hunk 4]
```diff
@@ -761,22 +789,19 @@
...
         "@tf.function\n",
         "def train_step(real_x, real_y):\n",
-        "  # persistent is set to True because gen_tape and disc_tape is used more than\n",
+        "  # persistent is set to True because gradient_tape is used more than\n",
         "  # once to calculate the gradients.\n",
-        "  with tf.GradientTape(persistent=True) as gen_tape, tf.GradientTape(\n",
-        "      persistent=True) as disc_tape:\n",
-        "    # Generator G translates X -\u003e Y\n",
-        "    # Generator F translates Y -\u003e X.\n",
+        "  with tf.GradientTape(persistent=True) as gradient_tape:\n",
```
Comment (yashk2810) [inline site/en/r2/tutorials/generative/cyclegan.ipynb]: Maybe use tape?

File: site/en/r2/tutorials/generative/cyclegan.ipynb (hunk 1 above)
Comment (junghau) [reply site/en/r2/tutorials/generative/cyclegan.ipynb]: Do you mean to revert back the image paths to original, as in the form `images/horse2zebra_1.png`?

Comment (yashk2810) [reply site/en/r2/tutorials/generative/cyclegan.ipynb]: The original image paths (images/horse2zebra_1.png) works fine. Why the change?

File: site/en/r2/tutorials/generative/cyclegan.ipynb (hunk 4 above)
Comment (junghau) [reply site/en/r2/tutorials/generative/cyclegan.ipynb]: Good idea.

**Comment by yashk2810** [comment]
LGTM
